packages/superset-ui-chart-controls/src/utils/advancedAnalytics/rollingWindow.ts

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable camelcase */
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitationsxw
+ * under the License.
+ */
+import {
+  ensureIsArray,
+  ensureIsInt,
+  QueryFormData,
+  QueryObject,
+  RollingType,
+} from '@superset-ui/core';
+
+export function rollingWindowTransform(
+  formData: QueryFormData,
+  queryObject: QueryObject,
+): QueryObject {
+  if (
+    formData.rolling_type === RollingType.None ||
+    !formData.rolling_type ||
+    formData.metrics === undefined
+  ) {
+    return queryObject;
+  }
+
+  // ensure `post_processing` is a copy from queryObject
+  const post_processing = ensureIsArray(queryObject.post_processing).slice();
+  const columns = Object.fromEntries(
```

Comment:
Seems like this could be made into a reusable function / this logic should be documented to understand conversion between formData.metrics and post_processing format.